### PR TITLE
Cleanup bank conflict checker to better infer launch param

### PR DESCRIPTION
**Pull request by zasdfgbnm**
- Use `kernel->summary().parallel_dimension_map_` to replace `InferLaunchParams`
- Use `LaunchParams()` (all `LaunchParams::UNINITIALIZED_VAL`) to replace `c10::optional<LaunchParams>`

### CODE REVIEW

File: csrc/lower_bank_conflict.cpp [hunk 1]
```diff
@@ -194,97 +197,67 @@ int getConflictWays(const std::vector<int64_t>& addresses) {
   return conflict;
 }
 
-class InferLaunchParams : public kir::IrVisitor {
```
Comment (zasdfgbnm) [inline csrc/lower_bank_conflict.cpp:197]: This class is replaced by `BankConflictInfo::inferLaunchParams`

File: csrc/lower_bank_conflict.h [hunk 2]
```diff
@@ -43,8 +43,8 @@ namespace nvfuser {
 // way == 0 --> not applicable (for example, tensor is not a smem tensor)
 // way == 1 --> no conflict
 std::unordered_map<const Expr*, std::pair<int, int>> getBankConflictInfo(
-    kir::Kernel* kernel,
-    c10::optional<LaunchParams> launch_params = c10::nullopt,
+    const kir::Kernel* kernel,
+    LaunchParams launch_params = {},
```
Comment (zasdfgbnm) [inline csrc/lower_bank_conflict.h:47]: `LaunchParams` supports uninitialized value, so it does not make sense to have `c10::optional<LaunchParams>`. Before this PR, `getBankConflictInfo` returns wrong result if `LaunchParams()` is provided as `launch_params`, this PR fixes this problem.

**Comment by zasdfgbnm** [comment]
!build

**Review by naoyam — APPROVED** [review]
LGTM

**Pull request merged**